src/main/java/net/fabricmc/loader/impl/FormattedException.java

```diff
@@ -44,7 +47,33 @@ public FormattedException(String mainText, Throwable cause) {
 		this.mainText = mainText;
 	}
 
+	public static FormattedException ofLocalized(String key, String message) {
+		return new FormattedException(Localization.formatRoot(key), message).addTranslation(key);
+	}
+
+	public static FormattedException ofLocalized(String key, String format, Object... args) {
+		return new FormattedException(Localization.formatRoot(key), format, args).addTranslation(key);
+	}
+
+	public static FormattedException ofLocalized(String key, String message, Throwable cause) {
+		return new FormattedException(Localization.formatRoot(key), message, cause).addTranslation(key);
+	}
+
+	public static FormattedException ofLocalized(String key, Throwable cause) {
+		return new FormattedException(Localization.formatRoot(key), cause).addTranslation(key);
+	}
+
 	public String getMainText() {
 		return mainText;
 	}
+
+	/* @Nullable */
```

Comment:
Not something for this PR, but we should really get these annotations added 👍

Reply:
Wait, this isn't nullable

Reply:
```suggestion
```